### PR TITLE
Fix gihub workflow for forked repo and dependabot

### LINKED ISSUE #292: Publish-npm-beta github action fails from fork and dependabot PR
PRs created from a fork or dependabot are causing failure in GH action [publish-npm-beta](https://github.com/kbss-cvut/s-forms/blob/dbf83e056824bbea513fbece4f79696516c92d1a/.github/workflows/npm-publish-beta.yml#L23) step. 

1) PR created from fork https://github.com/kbss-cvut/s-forms/pull/288 resulted in GH action failure https://github.com/kbss-cvut/s-forms/actions/runs/8362521150:
```
Run phips28/gh-action-bump-version@master.
...
...
runInWorkspace | command: git args: [ 'checkout', 'fix/imports' ]
✖  fatal     error: pathspec 'fix/imports' did not match any file(s) known to git

git exited with code 1
✖  fatal     Failed to bump version
````

2) Same failure occurred due to dependabot PR  https://github.com/kbss-cvut/s-forms/pull/290, which resulted in GH action
https://github.com/kbss-cvut/s-forms/actions/runs/8386115851/job/22965992150 containing failure:
```
Run phips28/gh-action-bump-version@master.
...
...
runInWorkspace | command: git args: [ 'checkout', 'dependabot/npm_and_yarn/webpack-dev-middleware-6.1.2' ]
✖  fatal     error: pathspec 'dependabot/npm_and_yarn/webpack-dev-middleware-6.1.2' did not match any file(s) known to git

git exited with code 1
✖  fatal     Failed to bump version
```

-----

Moreover, note we have some deprecations in build which **most-likely** does not cause the issue, but worth mention it here / can be solved as part of solution.
See https://github.com/kbss-cvut/s-forms/actions/runs/8362521150
image

### PR DESCRIPTION
**Pull request by LaChope**
Resolves #292.

I did not have the opportunity to test it, let's remember it for next time there is a merged PR from a forked repo.

### CODE REVIEW

File: .github/workflows/npm-publish-alpha.yml [hunk 1]
```diff
@@ -10,13 +10,13 @@ jobs:
...
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Get short SHA
-        id: sha
-        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+        id: set_sha
+        run: echo "SHA_SHORT=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
```
Comment (blcham) [inline .github/workflows/npm-publish-alpha.yml:19]: wow ... `>> $GITHUB_ENV` is quite strange. Wondering why this is needed?

File: .github/workflows/npm-publish-alpha.yml [hunk 2]
```diff
@@ -25,13 +25,13 @@ jobs:
           major-wording: "MAJOR RELEASE"
           patch-wording: "PATCH RELEASE"
           rc-wording: null
-          preid: ${{ format('alpha-{0}', steps.sha.outputs.sha_short) }}
+          preid: ${{ format('beta-{0}', env.SHA_SHORT) }}
```
Comment (blcham) [inline .github/workflows/npm-publish-alpha.yml:28]: I guess beta --> alpha ?

File: .github/workflows/npm-publish-beta.yml [hunk 3]
```diff
@@ -12,13 +12,13 @@ jobs:
...
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Get short SHA
-        id: sha
-        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+        id: set_sha
+        run: echo "SHA_SHORT=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
```
Comment (blcham) [inline .github/workflows/npm-publish-beta.yml:21]: see note above

**Review by blcham — CHANGES_REQUESTED** [review]
See my comments.

It needs to be tested right after merge.

File: .github/workflows/npm-publish-alpha.yml (hunk 1 above)
Comment (LaChope) [reply .github/workflows/npm-publish-alpha.yml:19]: This comes from this deprecation note: 

image

According to Github [documentation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files), we should use it like so: `echo "{environment_variable_name}={value}" >> "$GITHUB_ENV"`

File: .github/workflows/npm-publish-alpha.yml (hunk 2 above)
Comment (LaChope) [reply .github/workflows/npm-publish-alpha.yml:28]: Yes, mistake here

**Pull request merged**